sycl/include/sycl/ext/oneapi/matrix/matrix-unified.hpp

```diff
@@ -486,13 +485,23 @@ void joint_matrix_copy(
   std::ignore = sg;
   dst.matrix_impl.wi_marray = src.matrix_impl.wi_marray;
 #else
+  auto wi_data_c = sycl::ext::oneapi::detail::get_wi_data(sg, src);
+  auto wi_data_dst = sycl::ext::oneapi::detail::get_wi_data(sg, dst);
```

Comment:
I'd suggest moving these 2 lines back to where they were to minimize unnecessary changes in this PR.